Formula/trzsz.rb

```diff
@@ -0,0 +1,55 @@
+class Trzsz < Formula
+  include Language::Python::Virtualenv
+
+  desc "Simple file transfer tools, similar to lrzsz (rz/sz), and compatible with tmux"
+  homepage "https://trzsz.github.io"
+  url "https://files.pythonhosted.org/packages/50/01/b8679a72c0604c964a6c4f46f100fa707ecad97c34f61e58334f764e9cca/trzsz-0.3.1.tar.gz"
+  sha256 "97b008cd5441583e9090c9078854e90b3cdfe2fd95a1cfb717efee1b935a7737"
+  license "MIT"
+
+  depends_on "python@3.10"
+
+  resource "trzsz-libs" do
+    url "https://files.pythonhosted.org/packages/5e/c3/a5134c70082e80d402594b45e5d6a246f0e34a2fd5004bc83bb116a7ff14/trzsz-libs-0.3.1.tar.gz"
+    sha256 "bea662ff2f553aae057910dc7427b235370e9c822ec3a4ad27cb1220f98d5e8d"
+  end
+
+  resource "trzsz-svr" do
+    url "https://files.pythonhosted.org/packages/e9/7a/147baf7c62f25bc39c50dbdbad469da9536c592cc53de9cfb71c86256a1c/trzsz-svr-0.3.1.tar.gz"
+    sha256 "5e2334b402a968936df312fcfaf81628c1fae08e63aa086fbfd6adacf4b89257"
+  end
+
+  resource "trzsz-iterm2" do
+    url "https://files.pythonhosted.org/packages/3e/5b/81e0de0f52895439be6bde9e0152c4e9abb54eb5a2215e03972c47d378d8/trzsz-iterm2-0.3.1.tar.gz"
+    sha256 "75fefed756dc1577d84c14b0d3eac90050229063d1403f3f83bf0dcc8df8f0de"
+  end
+
+  def install
+    virtualenv_install_with_resources
+    bin.install_symlink libexec/"bin/trz"
+    bin.install_symlink libexec/"bin/tsz"
+    bin.install_symlink libexec/"bin/trzsz-iterm2"
+  end
+
+  def assert_trzsz_output(output, cmd)
+    Open3.popen3(cmd) do |_stdin, stdout, _stderr, wait_thr|
+      ohai cmd
+      sleep 1
+      begin
+        Process.kill(:INT, wait_thr.pid)
+      rescue
+        # do nothing
+      end
+      assert_match output, stdout.read
+    end
+  end
```

Comment:
This is much better; thanks.

Do we need a custom method here? Can we just use `shell_output`?

Reply:
The child process will hang in there, and wait for input.
We have to kill the child process.

Reply:
Let's just do this once so we don't need a special method for it then. You probably don't need the `rescue` either.

Reply:
would `pipe_output` work?

Reply:
> would `pipe_output` work?

Maybe. I will try it later.

Reply:
`pipe_output` is not working for me.

```rb
def pipe_output(cmd, input = nil, result = nil)
  output = IO.popen(cmd, "w+") do |pipe|
    sleep 1   # without the sleep, the test will be hang up.
    pipe.write(input) unless input.nil?
    pipe.close_write
    pipe.read
  end
  output
rescue Minitest::Assertion
  puts output
  raise
end

puts pipe_output "trz", "\n"
```

1. Have to sleep a while before writing the input.
2. The test environment doesn't support reseting `tty`. 
```
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/trzsz/svr/recv.py", line 76, in main
    tty.setraw(sys.stdin.fileno(), termios.TCSADRAIN)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/tty.py", line 20, in setraw
    mode = tcgetattr(fd)
error: (25, 'Inappropriate ioctl for device')
hello.rb:9:in `rescue in pipe_output': uninitialized constant Minitest (NameError)

rescue Minitest::Assertion
       ^^^^^^^^
	from hello.rb:1:in `pipe_output'
	from hello.rb:14:in `<main>'
hello.rb:4:in `write': Broken pipe (Errno::EPIPE)
	from hello.rb:4:in `block in pipe_output'
	from hello.rb:2:in `popen'
	from hello.rb:2:in `pipe_output'
	from hello.rb:14:in `<main>'
```
3. There will be an output `::TRZSZ:TRANSFER:R:` before reset tty.
4. In case the test hangs, I try to kill the child process.

Reply:
I changed to test failure cases.